src/contracts/BCoWPool.sol

```diff
@@ -142,11 +142,14 @@ contract BCoWPool is IERC1271, IBCoWPool, BPool, BCoWConst {
   /**
    * @inheritdoc BPool
    * @dev Grants infinite approval to the vault relayer for all tokens in the
-   * pool after the finalization of the setup.
+   * pool after the finalization of the setup. Also emits COWAMMPoolCreated() event.
    */
   function _afterFinalize() internal override {
     for (uint256 i; i < _tokens.length; i++) {
       IERC20(_tokens[i]).approve(VAULT_RELAYER, type(uint256).max);
     }
+
+    // Emit event
+    IBCoWFactory(_factory).logBCoWPool();
```

Comment:
it's either assert in the constructor that `_factory.code.length > 0`, or we implement a try-catch here, i'd incline for the latter, but this would break the whole pool configuration if it's standalone deployed.

i'm thinking that, if reverts, we could emit the event in this context:
```solidity
try {IBCoWFactory(_factory).logBCoWPool();}
catch {emit IBCoWFactory.COWAMMPoolCreated(address(this);}
```

wdyt of that?
i'd add a comment explaining that we try to emit on the factory for simplicity of indexing

Reply:
hmm we dont have the event in that context, imho will should just return. If a pool not deployed by factory wants to index a finalize we can create a `event PoolFinalized();`